packages/core/src/bodl/bodl-emitter-service.ts

```diff
@@ -0,0 +1,187 @@
+import { LineItemMap } from '../cart';
+import { Checkout, CheckoutService } from '../checkout';
+import { ShopperCurrency } from '../config';
+import { Order } from '../order';
+
+import BodlService from "./bodl-service";
+import { BodlEventsCheckout } from './bodl-window';
+
+export default class BodlEmitterService implements BodlService {
+    private _checkoutStarted = false;
+
+    constructor(
+        private checkoutService: CheckoutService,
+        private bodlEvents: BodlEventsCheckout
+    ) {
+
+    }
+
+    checkoutBegin(): void {
+        if (this._checkoutStarted) {
+            return;
+        }
+
+        const checkout = this.getCheckout();
```

Comment:
```suggestion
        const checkout = this.getCheckoutOrThrow();
```

Reply:
That way you can avoid the if condition in the next line

Reply:
Also if you just need cart data, does it make sense to just do `const cart = this.getCartOrThrow();`

Reply:
What this method will do? I don't want to break  the application (throw an error) if there is no available data.